app/controllers/cinema_halls_controller.rb

```diff
@@ -14,7 +14,10 @@ def show
   def create
     cinema_hall = CinemaHall.new(cinema_hall_params)
     if [20, 50, 100, 200].include? cinema_hall.size
-      render json: cinema_hall, status: :created if cinema_hall.save
+      if cinema_hall.save
+        render json: cinema_hall, status: :created
+        CinemaHallsService.new(self).assign_seat_names
```

Comment:
I believe that you have to send `cinema_hall` to service instead of self. 
A few words about [self](https://www.rubyguides.com/2020/04/self-in-ruby/)